[TT-16945] docs: Tyk Pump 1.15.0 release notes (#1825)
TykTechnologies/tyk-docs: developer-support/release-notes/pump.mdx

</Accordion>

<Accordion title='Add MCP analytics support'>
Tyk Pump 1.15.0 adds support for storing and processing MCP (Model Context Protocol) Gateway analytics records:

- **Storage backends**: MCP analytics records are supported across MongoDB, PostgreSQL, Elasticsearch, and Prometheus backends. Each record includes MCP-specific fields: `mcp_method`, `mcp_primitive_type`, `mcp_primitive_name`, and `mcp_error_code`.
- **Hybrid pump aggregation**: A new `EnableMCPAggregation` flag on the hybrid pump enables pre-aggregation of MCP records before they are forwarded to the Control Plane, reducing data volume for high-traffic deployments.
- **Prometheus labels**: Three new Prometheus labels are emitted for MCP traffic: `mcp_method`, `mcp_primitive_type`, and `mcp_primitive_name`, aligned with the OpenTelemetry `mcp.*` semantic conventions. A new `mcp_only` filter restricts Prometheus output to MCP records exclusively.

> - **Bug fix**: Resolved a duplicate log entry issue where MCP records were being written twice to analytics storage.

Reviewer: ```suggestion

```
@JoanCamosTyk @v008370-git I think we should remove this

Reviewer 2: Agreed - this is N/A

#### Changelog
<a id="Changelog-v1.15.0" data-scroll-offset></a>

##### Added

<AccordionGroup>

<Accordion title='Improve readability of stdout pump JSON logs'>
Enhanced the stdout pump to output cleaner, more readable JSON logs by default. Escaped characters such as newlines and carriage returns are now properly formatted in the raw request and response fields, making the logs easier to read and analyze. A new configuration option `use_legacy_payload_format` has been added to revert to the previous behavior if needed.
</Accordion>

<Accordion title='Add MCP analytics support'>
Tyk Pump 1.15.0 adds support for storing and processing MCP (Model Context Protocol) Gateway analytics records:

- **Storage backends**: MCP analytics records are supported across MongoDB, PostgreSQL, Elasticsearch, and Prometheus backends. Each record includes MCP-specific fields: `mcp_method`, `mcp_primitive_type`, `mcp_primitive_name`, and `mcp_error_code`.
- **Hybrid pump aggregation**: A new `EnableMCPAggregation` flag on the hybrid pump enables pre-aggregation of MCP records before they are forwarded to the Control Plane, reducing data volume for high-traffic deployments.

> - **Prometheus labels**: Three new Prometheus labels are emitted for MCP traffic: `mcp_method`, `mcp_primitive_type`, and `mcp_primitive_name`, aligned with the OpenTelemetry `mcp.*` semantic conventions. A new `mcp_only` filter restricts Prometheus output to MCP records exclusively.
> 

Reviewer: ```suggestion
- **Prometheus labels**: Three new Prometheus labels are emitted for MCP traffic: `mcp_method`, `mcp_primitive_type`, and `mcp_primitive_name`, aligned with the OpenTelemetry `mcp.*` semantic conventions. A new `mcp_only` filter restricts Prometheus output to MCP records exclusively.
- **Bug fix**: Resolved a duplicate log entry issue where MCP records were being written twice to analytics storage.
```

For details, see the [MCP analytics documentation](/ai-management/mcp-gateway/mcp-analytics).
</Accordion>

</AccordionGroup>

##### Fixed

<AccordionGroup>

<Accordion title='Fix duplicate MCP analytics log entries'>
Resolved an issue where MCP records were being written twice to analytics storage. Tyk Pump will now correctly write each MCP analytics record once.

> </Accordion>

Reviewer: ```suggestion
```

---

## 1.15 Release Notes

### 1.15.0 Release Notes

#### Release Date 08 May 2026

#### Release Highlights

> In this release, we have improved the stdout pump with cleaner, more readable JSON logs by properly formatting escaped characters in request and response fields. We have also introduced MCP analytics support, enabling Tyk Pump to store and process MCP Gateway traffic records across MongoDB, PostgreSQL, Elasticsearch, and Prometheus backends.

Reviewer 2: Refocus this - the highlight is adding support for MCP.

Ensure you add a link to /ai-management/mcp-gateway/managing-mcp-proxies

For a comprehensive list of changes, please refer to the detailed [changelog](#Changelog-v1.15.0). 

#### Breaking Changes

**stdout pump JSON log formatting has changed by default**

> The stdout pump now outputs cleaner JSON logs by default: escaped newline (`\n`), tab (`\t`), and carriage return (`\r`) characters are no longer present in the `raw_request` and `raw_response` fields. While the output remains valid JSON, customers whose log-processing pipelines, dashboards, or alerting rules rely on the previous escaped-character format will need to either update their tooling to handle the new format or opt back into the previous behaviour.

Reviewer 2: ```suggestion
The **stdout pump** now outputs cleaner JSON logs. Newline (`\n`), tab (`\t`), and carriage return (`\r`) characters are no longer escaped in the `raw_request` and `raw_response` fields.

Users whose log-processing pipelines, dashboards, or alerting rules rely on the previous escaped-character format will need to either update their tooling to handle the new format or set the new [`pumps.stdout.use_legacy_payload_format`](/tyk-pump/tyk-pump-configuration/tyk-pump-environment-variables#pumps-stdout-use_legacy_payload_format) option to `true` in their Pump configuration.
```

> To preserve the previous behaviour, set the following in your stdout pump configuration:

Reviewer 2: ```suggestion
```

> 

Reviewer 2: ```suggestion
```

> `use_legacy_payload_format: true`

Reviewer 2: ```suggestion
```

> When set to `true`, the stdout pump output is identical to previous releases. When unset or set to `false`, the new clean formatting is applied.

Reviewer 2: ```suggestion
When `pumps.stdout.use_legacy_payload_format: true`, the stdout pump output is identical to previous releases. When unset or set to `false`, the new clean formatting is applied.
```

#### Dependencies

##### 3rd Party Dependencies & Tools

| Third Party Dependency                                    | Tested Versions   | Compatible Versions      |
| :--------------------------------------------------------- | :----------------- | :------------------------ | 
| [MongoDB](https://www.mongodb.com/try/download/community) | 5.x, 6.x, and 7.0 | 4.4.x, 5.x, 6.x, and 7.0 |
| [PostgreSQL](https://www.postgresql.org/download/)        | 13.x - 17.x       | 13.x - 17.x              | 
| [Redis](https://redis.io/download/)                       | 6.x - 7.0         | 6.x - 7.x                | 
| [Valkey](https://valkey.io/download/)                     | 8.0.x, 8.1.x      | 7.2.x, 8.0.x, 8.1.x      |                            

Given the time difference between your upgrade and the release of this version, we recommend customers verify the ongoing support of third-party dependencies they install, as their status may have changed since the release.

#### Deprecations

There are no deprecations in this release.

#### Upgrade instructions

For users currently on v1.14.1, we strongly recommend promptly upgrading to the latest release.

> In hybrid deployments, MDCB 2.11.0 must be deployed before upgrading Pump to 1.15.0. Keep `enable_mcp_aggregation` set to `false` during the upgrade — only set it to `true` once MDCB, Pump, Gateway, and Dashboard are all on their new versions. Setting it before MDCB is upgraded will cause RPC failures and silent loss of MCP analytics data. See [MCP Gateway upgrade considerations](/developer-support/upgrading#mcp-upgrade-considerations) for the full enablement sequence.

Reviewer 2: What do we mean by "hybrid deployments" here? This is a potentially ambiguous term so we must be precise - and link to the definition.

Go to the [Upgrading Tyk](#upgrading-tyk) section for detailed upgrade Instructions.

#### Downloads
- [Docker Image v1.15.0](https://hub.docker.com/r/tykio/tyk-pump-docker-pub/tags?page=&page_size=&ordering=&name=v1.15.0)
  - ```bash
    docker pull tykio/tyk-pump-docker-pub:v1.15.0
    ```
- Source code tarball for OSS - [GH Tyk Pump Repo](https://github.com/TykTechnologies/tyk-pump/releases/tag/v1.15.0)

#### Changelog
<a id="Changelog-v1.15.0" data-scroll-offset></a>

##### Changed

<AccordionGroup>

> <Accordion title='Cleaner JSON formatting for stdout pump logs by default'>

Reviewer 2: ```suggestion
<Accordion title='Cleaner JSON formatting for stdout pump logs'>
```

> The stdout pump now outputs cleaner, more readable JSON logs by default. Escaped newline, tab, and carriage return characters are no longer present in the `raw_request` and `raw_response` fields, making the logs significantly easier to read and analyse while remaining valid JSON. A new `use_legacy_payload_format` configuration option (default `false`) is available to revert to the previous escaped-character output for customers whose downstream tooling depends on the original format. This is a breaking change in default behaviour, see the `Breaking Changes` section for details.

Reviewer 2: ```suggestion
The **stdout pump** now outputs cleaner JSON logs, making the logs significantly easier to read and analyze. Newline (`\n`), tab (`\t`), and carriage return (`\r`) characters are no longer escaped in the `raw_request` and `raw_response` fields.

This is a breaking change for any user whose downstream tooling depends on the original format. A new `pumps.stdout.use_legacy_payload_format` configuration option (default `false`) is available if you need to continue to use the previous escaped-character output..
```

</Accordion>

</AccordionGroup>

##### Added

<AccordionGroup>

<Accordion title='Add MCP analytics support'>

> Tyk Pump 1.15.0 adds support for storing and processing MCP (Model Context Protocol) Gateway analytics records:

Reviewer 2: ```suggestion
This release adds support for the storing and processing of MCP (Model Context Protocol) Gateway analytics records:
```

- **Storage backends**: MCP analytics records are supported across MongoDB, PostgreSQL, Elasticsearch, and Prometheus backends. Each record includes MCP-specific fields: `mcp_method`, `mcp_primitive_type`, `mcp_primitive_name`, and `mcp_error_code`.

> - **Hybrid pump aggregation**: A new `EnableMCPAggregation` flag on the hybrid pump enables pre-aggregation of MCP records before they are forwarded to the Control Plane, reducing data volume for high-traffic deployments.

Reviewer 2: ```suggestion
- **Hybrid pump aggregation**: A new [`meta.enable_mcp_aggregation`](/tyk-pump/tyk-pump-configuration/tyk-pump-environment-variables#pumps-hybrid-meta-enable_mcp_aggregation) configuration option for the **hybrid** Pump enables pre-aggregation of MCP records before they are forwarded to the Control Plane, reducing data volume for high-traffic deployments.
```

Please confirm the actual config option name - we do not use camelCase in the component configurations. I think it is pumps.hybrid.meta.enable_mcp_aggregation but this must be checked and updated in the text. (ref: https://github.com/TykTechnologies/tyk-docs/pull/2024/changes#diff-51efc97fdce62d09ba0cc22b911842d85689e5879d3598a13c52a5cb1243c71eR1372)

---

## 1.15 Release Notes

### 1.15.0 Release Notes

#### Release Date 19 May 2026

#### Release Highlights

> Tyk Pump 1.15.0 introduces **MCP analytics support**, bringing full observability to [MCP Proxy](/ai-management/mcp-gateway/managing-mcp-proxies) traffic. MCP analytics records are captured and processed across MongoDB, PostgreSQL, Elasticsearch, and Prometheus backends, with no changes required to existing Pump configurations.

Reviewer: ```suggestion
Tyk Pump 1.15.0 introduces **MCP analytics support**, bringing full observability to [MCP Proxy](/ai-management/mcp-gateway/managing-proxies) traffic. MCP analytics records are captured and processed across MongoDB, PostgreSQL, Elasticsearch, and Prometheus backends, with no changes required to existing Pump configurations.
```

> Tyk Pump 1.15.0 introduces **MCP analytics support**, bringing full observability to [MCP Proxy](/ai-management/mcp-gateway/managing-mcp-proxies) traffic. MCP analytics records are captured and processed alongside HTTP API traffic across MongoDB, PostgreSQL, Elasticsearch, and Prometheus backends, with no changes required to existing pump configurations.

Reviewer 2: ```suggestion
Tyk Pump 1.15.0 introduces **MCP analytics support**, bringing full observability to [MCP Proxy](/ai-management/mcp-gateway/managing-mcp-proxies) traffic. MCP analytics records are captured and processed across MongoDB, PostgreSQL, Elasticsearch, and Prometheus backends, with no changes required to existing Pump configurations.
```

> This release also improves stdout pump output with cleaner, more readable JSON logs by properly formatting escaped characters in request and response fields.

Reviewer 2: ```suggestion
This release also improves **Stdout Pump** output with cleaner, more readable JSON logs by properly formatting escaped characters in request and response fields.
```

For a comprehensive list of changes, please refer to the detailed [changelog](#Changelog-v1.15.0).

#### Breaking Changes

> **stdout pump JSON log formatting has changed by default**

Reviewer 2: ```suggestion
**Stdout Pump JSON log formatting has changed by default**
```

> The **stdout pump** now outputs cleaner JSON logs. Newline (`\n`), tab (`\t`), and carriage return (`\r`) characters are no longer escaped in the `raw_request` and `raw_response` fields.

Reviewer 2: ```suggestion
The **Stdout Pump** now outputs cleaner JSON logs. Newline (`\n`), tab (`\t`), and carriage return (`\r`) characters are no longer escaped in the `raw_request` and `raw_response` fields.
```

Users whose log-processing pipelines, dashboards, or alerting rules rely on the previous escaped-character format will need to either update their tooling to handle the new format or set the new [`pumps.stdout.use_legacy_payload_format`](/tyk-pump/tyk-pump-configuration/tyk-pump-environment-variables#pumps-stdout-use_legacy_payload_format) option to `true` in their Pump configuration.

> When `pumps.stdout.use_legacy_payload_format: true`, the stdout pump output is identical to previous releases. When unset or set to `false`, the new clean formatting is applied.

Reviewer 2: ```suggestion
When `pumps.stdout.use_legacy_payload_format: true`, the Stdout Pump output is identical to previous releases. When unset or set to `false`, the new clean formatting is applied.
```

#### Deprecations

There are no deprecations in this release.

#### Upgrade instructions

For users currently on v1.14.1, we strongly recommend promptly upgrading to the latest release.

In distributed deployments, MDCB 2.11.0 must be deployed before upgrading Pump to 1.15.0. Keep `enable_mcp_aggregation` set to `false` during the upgrade — only set it to `true` once MDCB, Pump, Gateway, and Dashboard are all on their new versions. Setting it before MDCB is upgraded will cause RPC failures and silent loss of MCP analytics data. See [MCP Gateway upgrade considerations](/developer-support/upgrading#mcp-upgrade-considerations) for the full enablement sequence.

Go to the [Upgrading Tyk](#upgrading-tyk) section for detailed upgrade Instructions.

#### Downloads
- [Docker Image v1.15.0](https://hub.docker.com/r/tykio/tyk-pump-docker-pub/tags?page=&page_size=&ordering=&name=v1.15.0)
  - ```bash
    docker pull tykio/tyk-pump-docker-pub:v1.15.0
    ```
- Source code tarball for OSS - [GH Tyk Pump Repo](https://github.com/TykTechnologies/tyk-pump/releases/tag/v1.15.0)

#### Changelog
<a id="Changelog-v1.15.0" data-scroll-offset></a>

##### Changed

<AccordionGroup>

> <Accordion title='Cleaner JSON formatting for stdout pump logs'>

Reviewer 2: ```suggestion
<Accordion title='Cleaner JSON formatting for Stdout Pump logs'>
```

> The **stdout pump** now outputs cleaner JSON logs, making the logs significantly easier to read and analyze. Newline (`\n`), tab (`\t`), and carriage return (`\r`) characters are no longer escaped in the `raw_request` and `raw_response` fields.

Reviewer 2: ```suggestion
The **Stdout Pump** now outputs cleaner JSON logs, making the logs significantly easier to read and analyze. Newline (`\n`), tab (`\t`), and carriage return (`\r`) characters are no longer escaped in the `raw_request` and `raw_response` fields.
```

This is a breaking change for any user whose downstream tooling depends on the original format. A new `pumps.stdout.use_legacy_payload_format` configuration option (default `false`) is available if you need to continue to use the previous escaped-character output..
</Accordion>

</AccordionGroup>

##### Added

<AccordionGroup>

<Accordion title='Add MCP analytics support'>
This release adds support for the storing and processing of MCP (Model Context Protocol) Gateway analytics records:

- **Storage backends**: MCP analytics records are supported across MongoDB, PostgreSQL, Elasticsearch, and Prometheus backends. Each record includes MCP-specific fields: `mcp_method`, `mcp_primitive_type`, `mcp_primitive_name`, and `mcp_error_code`.

> - **Hybrid pump aggregation**: A new [`pumps.hybrid.meta.enable_mcp_aggregation`](/tyk-pump/tyk-pump-configuration/tyk-pump-environment-variables#pumps-hybrid-meta-enable_mcp_aggregation) configuration option on the hybrid pump enables pre-aggregation of MCP records before they are forwarded to the Control Plane, reducing data volume for high-traffic distributed deployments.

Reviewer 2: ```suggestion
- **Hybrid Pump aggregation**: A new [`pumps.hybrid.meta.enable_mcp_aggregation`](/tyk-pump/tyk-pump-configuration/tyk-pump-environment-variables#pumps-hybrid-meta-enable_mcp_aggregation) configuration option on the Hybrid Pump enables pre-aggregation of MCP records before they are forwarded to the Control Plane, reducing data volume for high-traffic distributed deployments.
```

- **Prometheus labels**: Three new Prometheus labels are emitted for MCP traffic: `mcp_method`, `mcp_primitive_type`, and `mcp_primitive_name`, aligned with the OpenTelemetry `mcp.*` semantic conventions. A new `mcp_only` filter restricts Prometheus output to MCP records exclusively.

> For details, see the [MCP analytics documentation](/ai-management/mcp-gateway/mcp-analytics).

Reviewer: ```suggestion
For details, see the [MCP analytics documentation](/ai-management/mcp-gateway/mcp-observability).
```
